Update `precommit:emoji` script after GitHub's sunset of SVN

desrosj opened the pull request:
All build script related changes form #5804 have been copied over.

Trac ticket: https://core.trac.wordpress.org/ticket/60520

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

desrosj committed:
Include changes from #5804.

kraftbj committed:
Update file permissions for the shell script

kraftbj commented:
Tested. Only thing I ran into was permission needing updating on the shell script. Submitted a PR to add that via https://github.com/desrosj/wordpress-develop/pull/153

desrosj committed:
Merge pull request #153 from kraftbj/update/precommit-emoji

Update file permissions for the shell script

desrosj commented:
Thanks @kraftbj! I've merged that.

To bring a brief summary here, here are a few things I don't love about the current implementation (though not strongly opposed to):
- I don't love adding `gh` as a requirement for the script.
- I would prefer that the necessary script be included in `Grunt.js` instead of a separate directory for shell scripts (where this will be the only one).

kraftbj committed:
Add script directly to Gruntfile.js

kraftbj commented:
I don't love adding `gh` either, but given that we _do_ use GitHub as a partner service extensively already, it seems okay for now to get this out the door.

I tried to make it all part of the Grunt JS file in https://github.com/desrosj/wordpress-develop/pull/154 -- can you give that a spin?

peterwilsoncc commented on Gruntfile.js:
```diff
@@ -332,7 +332,6 @@ module.exports = function(grunt) {
 					[ WORKING_DIR + 'wp-admin/js/tags-suggest.js' ]: [ './src/js/_enqueues/admin/tags-suggest.js' ],
 					[ WORKING_DIR + 'wp-admin/js/tags.js' ]: [ './src/js/_enqueues/admin/tags.js' ],
 					[ WORKING_DIR + 'wp-admin/js/site-health.js' ]: [ './src/js/_enqueues/admin/site-health.js' ],
-					[ WORKING_DIR + 'wp-admin/js/site-icon.js' ]: [ './src/js/_enqueues/admin/site-icon.js' ],
```
Is this intentional, it looks unrelated.

desrosj committed:
Merge pull request #154 from kraftbj/update/precommit-emoji

Add script directly to Gruntfile.js

desrosj commented:
Thanks all for keeping this going! I think we're at a good spot. There's just one outstanding question about what seems like an unintentional deleted line.

kraftbj commented:
I concur. If the deletion was intentional (I'm not sure why it would be), it should be noted in the commit to explain the connection. But, with that line restored, LGTM.

desrosj committed:
Include `precommit:emoji` in the build process test workflow.

desrosj committed:
Pass a GH token.

desrosj committed:
Re-add accidental change.

desrosj committed:
Revert last Twemoji update for build process testing.

desrosj committed:
Small change. GH did not sync last commit for some reason.

desrosj committed:
Revert "Small change. GH did not sync last commit for some reason."

This reverts commit 6477e5ffd1b321e3530f73396b61bc1307a86939.

desrosj commented:
It was me that removed that line, wasn't it? Just realized that looking through this again. 🙃

Since I can't recall why I included that, or see a reason how it's related, I've gone and removed it.

I also added a new step to the Actions workflow that tests our build process to run the `precommit:emoji` script to confirm all expected changes are included. GitHub is not syncing pushes right now for some reason. But once I confirm it correctly flags missing changes, I'll get this merged.

peterwilsoncc commented:
> I also added a new step to the Actions workflow that tests our build process to run the precommit:emoji script to confirm all expected changes are included. GitHub is not syncing pushes right now for some reason. But once I confirm it correctly flags missing changes, I'll get this merged.

As the script currently reads the files from the `main` branch, this may need to use something version specific so it doesn't trigger build failures due to modified files as the folder is updated upstream.

desrosj committed:
Pin a specific tag to the build process.

This prevents upstream changes from breaking our test scripts.

swissspidy commented on .github/workflows/test-build-processes.yml:
```diff
@@ -31,7 +31,7 @@ jobs:
   # Tests the WordPress Core build process on multiple operating systems.
   test-core-build-process:
     name: Core running from ${{ matrix.directory }}
-    uses: WordPress/wordpress-develop/.github/workflows/callable-test-core-build-process.yml@trunk
```
To undo before commit

desrosj commented:
> As the script currently reads the files from the `main` branch, this may need to use something version specific so it doesn't trigger build failures due to modified files as the folder is updated upstream.

I've tested using a tag reference instead of branch, and it seems to work alright. Talked through this a bit in Slack with @peterwilsoncc and since this adds an additional spot to update when updating the library, it would be nice to automate it a bit.

desrosj commented:
Merged in https://core.trac.wordpress.org/changeset/57758.
